static/app/utils/metrics/dashboardImport.tsx

```diff
@@ -202,42 +203,54 @@ export class WidgetParser {
   private parseRequest(request: Request) {
     const {queries, formulas = [], response_format, display_type} = request;
 
-    const parsedFormulas = formulas.map(f => this.parseFormula(f));
-
     const parsedQueries = queries
-      .filter(q => parsedFormulas.includes(q.name))
-      .map(q => this.parseQuery(q));
+      .map(query => this.parseQuery(query))
+      .sort((a, b) => a!.name.localeCompare(b!.name));
 
     if (response_format !== 'timeseries') {
       this.errors.push(
         `widget.request.response_format - unsupported: ${response_format}`
       );
     }
 
+    const equationFormulas = formulas.filter(f =>
+      // indicates a more complex formula and not just a reference to a query
+      f.formula.trim().includes(' ')
+    );
+    const parsedEquations = this.parseEquations(parsedQueries, equationFormulas);
+
     const displayType = this.parseDisplayType(display_type);
 
     this.parseStyle(request.style);
 
     return {
       displayType,
       queries: parsedQueries,
+      equations: parsedEquations,
     };
   }
 
-  private parseFormula(formula: Formula) {
-    if (!formula.formula.includes('(')) {
-      return formula.formula;
-    }
+  // swaps query names with query symbols in formulas eg. query1 + $query0 => $b + $a
+  private parseEquations(queries: any[], formulas: Formula[]) {
+    const queryNames = queries.map(q => q.name);
+    const queryNameMap = queries.reduce((acc, query, index) => {
+      acc[query.name] = getQuerySymbol(index);
+      return acc;
+    }, {});
 
-    const [functionName, ...args] = formula.formula
-      .split(/\(|\)|,/)
-      .filter(Boolean)
-      .map(s => s.trim());
+    const equations = formulas.map(formula => {
+      const {formula: formulaString, alias} = formula;
+      const mapped = queryNames.reduce((acc, queryName) => {
+        return acc.replace(queryName, `$${queryNameMap[queryName]}`);
```

Comment:
In case a query is used multiple times?
```suggestion
        return acc.replaceAll(queryName, `$${queryNameMap[queryName]}`);
```